### ISSUE TITLE
gifload_buffer: bad frame size

### ISSUE THREAD

**Issue by legopin** [libvips/libvips#1701]
## Expected Behavior
vips should copy GIF with no error

## Actual Behavior

<details>
<summary>sample Image Link (Warning lingerie photo)</summary>

![problem](https://user-images.githubusercontent.com/15863509/86338762-4acf6700-bc85-11ea-84f9-2ca2496a6863.gif)

</details>

When attempting to process GIF file, encounters error
```sh
vips copy problem.gif[n=-1] out.gif
(vips:58322): VIPS-WARNING **: 16:51:38.676: error in tile 0 x 1600
(vips:58322): VIPS-WARNING **: 16:51:38.677: error in tile 0 x 1616
(vips:58322): VIPS-WARNING **: 16:51:38.677: error in tile 0 x 1632
gifload: bad frame size
gifload: Failed to read from given file
gifload: Wrong record type detected
gifload: Wrong record type detected
gifload: Wrong record type detected
```

## Potential Source of error
The source of the error is here, however I am not sure how the sample image broke this part.
https://github.com/libvips/libvips/blob/3428767a464ac5ca46aa3d036c8c59152d9a934b/libvips/foreign/gifload.c#L521-L531

## Environment
vips-8.9.2-Tue Apr 21 09:26:11 UTC 2020
MacOS 10.14.6

**Comment by jcupitt** [comment]
Hello @legopin,

I had a look and this GIF has frames that are larger than the canvas:

```
john@kiwi:~/pics$ vipsheader -a lingere.gif 
vips_foreign_load_gif_header: 0x55e5e9f391a0
vips_foreign_load_gif_open_giflib:
vips_foreign_load_gif_scan:
vips_foreign_load_gif_scan_image: frame of 800x800 pixels at 0x0
vips_foreign_load_gif_scan_image: frame of 1106x1106 pixels at 0x0
vips_foreign_load_gif_scan_image: frame of 800x800 pixels at 0x0
vips_foreign_load_gif_scan_image: frame of 1105x1106 pixels at 0x0
vips_foreign_load_gif_scan_image: frame of 800x800 pixels at 0x0
vips_foreign_load_gif_scan_image: frame of 800x800 pixels at 0x0
vips_foreign_load_gif_close_giflib:
```

So 6 frames, the canvas is 800x800, but frames 2 and 4 are larger.

It looks like most viewers clip off these extra pixels, I'll add this feature to libvips too.

**Comment by jcupitt** [comment]
OK, this should now work in libvips master, thanks for the report! It will be included in libvips 8.10.

Any testing would be great.

**Comment by jcupitt** [comment]
beta2 of 8.10 is up now and fixes this: https://github.com/libvips/libvips/releases/tag/v8.10.0-beta2

I'll close.
